internal/cri/server/podsandbox/opts.go

```diff
@@ -32,20 +33,35 @@ func WithNRISandboxDelete(sandboxID string) containerd.ProcessDeleteOpts {
 		if !ok {
 			return nil
 		}
-		nric, err := nri.New()
-		if err != nil {
-			log.G(ctx).WithError(err).Error("unable to create nri client")
-			return nil
-		}
-		if nric == nil {
-			return nil
-		}
-		sb := &nri.Sandbox{
-			ID: sandboxID,
-		}
-		if _, err := nric.InvokeWithSandbox(ctx, task, v1.Delete, sb); err != nil {
-			log.G(ctx).WithError(err).Errorf("Failed to delete nri for %q", task.ID())
-		}
+
+		nriv010Invoke(ctx, task, sandboxID, v1.Delete, nil)
+
+		return nil
+	}
+}
+
+//nolint:staticcheck
+func nriv010Invoke(ctx context.Context, task containerd.Task, id string, state v1.State, labels map[string]string) error {
+	nric, err := nri.New()
+	if err != nil {
+		log.G(ctx).WithError(err).Error("unable to create nri client")
 		return nil
 	}
+	if nric == nil {
+		return nil
+	}
+
+	if plugins := nric.Plugins(); len(plugins) != 0 {
+		log.G(ctx).Warnf("Deprecated NRI 0.1.0-style plugins configured: %s", strings.Join(plugins, ","))
```

Comment:
I opened a PR for this previously here, in case we can draw inspiration from it:
https://github.com/containerd/containerd/pull/11951

My idea here was to emit a deprecation warning like we did for the other deprecated features.